### PR TITLE
Improve save_csv string formatting

### PR DESCRIPTION
**Pull request by bhagemeier**
Using .format can take up to 2x as long as using %.

Also add a test covering an additional line of code.

## Description

The title says it all. Improve write performance.

Issue/s resolved: #947 

## Changes proposed:
- Use string interpolation operator rather than str.format()

## Type of change
- performance improvement

## Memory requirements
- no changes

## Performance
Up to 2x as fast in overall runtime than using str.format(), e.g. cutting down from 139s to 66s for the same data. See performance comparison on #947 .

## Due Diligence
- [x] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: heat/core/io.py [hunk 1]
```diff
@@ -1004,16 +1004,16 @@ def save_csv(
         decimals = 0
         dec_sep = 0
         if sign == 1:
-            fmt = "{: %dd}" % (pre_point_digits + 1)
```
Comment (bhagemeier) [inline heat/core/io.py:1007]: Format strings are different depending on whether you use str.format() or string interpolation. String interpolation is significantly faster than str.format(). Therefore, this PR switches to string interpolation.

File: heat/core/io.py [hunk 2]
```diff
@@ -1033,11 +1033,11 @@ def save_csv(
     for i in range(data.lshape[0]):
         # if lshape is of the form (x,), then there will only be a single element per row
         if len(data.lshape) == 1:
-            row = fmt.format(data.larray[i])
+            row = fmt % (data.larray[i])
```
Comment (bhagemeier) [inline heat/core/io.py:1036]: The format strings have to be used differently now.

File: heat/core/tests/test_io.py [hunk 3]
```diff
@@ -195,6 +195,21 @@ def test_save_csv(self):
                             if data.comm.rank == 0:
                                 os.unlink(filename)
 
+        data = ht.random.randint(0, 100, size=(150,))
```
Comment (bhagemeier) [inline heat/core/tests/test_io.py:198]: This captures a special case that had not been covered before.

File: heat/core/io.py (hunk 1 above)
Comment (mtar) [reply heat/core/io.py:1007]: Have you tried f-strings? How do they compete?

Comment (mtar) [reply heat/core/io.py:1007]: Have you tried f-strings? How do they compete?

Comment (bhagemeier) [reply heat/core/io.py:1007]: Yes, tried them. Seems a bit convoluted to create them. f-strings are intended to be in code as literals, whereas I dynamically generate the format string depending on desired precision etc. You can give it a try if you like. For me personally, it wasn't quite worth the effort for an output format that is far from ideal in terms of performance anyway.

**Comment by ClaudiaComito** [comment]
run tests

**Review by ClaudiaComito — APPROVED** [review]
Thanks a lot @bhagemeier !

**Pull request merged**